pkg/registry/node/strategy.go

```diff
@@ -162,9 +162,16 @@ func ResourceLocation(getter ResourceGetter, connection client.ConnectionInfoGet
 	}
 	host := hostIP.String()
 
-	if portReq == "" || strconv.Itoa(ports.KubeletPort) == portReq {
-		// Ignore requested scheme, use scheme provided by GetConnectionInfo
-		scheme, port, kubeletTransport, err := connection.GetConnectionInfo(host)
+	// We check if we want to get a default Kubelet's transport. It happens if either:
+	// - no port is specified in request (Kubelet's port is default),
+	// - we're using Port stored as a DaemonEndpoint and requested port is a Kubelet's port stored in the DaemonEndpoint,
+	// - there's no information in the API about DaemonEnpoint (legacy cluster) and requested port is equal to ports.KubeletPort (cluster-wide config)
+	kubeletPort := node.Status.DaemonEndpoints.KubeletEndpoint.Port
+	if kubeletPort == 0 {
+		kubeletPort = ports.KubeletPort
+	}
+	if portReq == "" || strconv.Itoa(kubeletPort) == portReq {
```

Comment:
hopefully anyone using the API proxy to reach nodes was not specifying a port... anyone who was specifying 10250 against < 1.1 would now be broken against nodes with other ports... not much we can do about that, I guess :-/